core/app/models/workarea/checkout/collect_payment.rb

```diff
@@ -36,12 +36,12 @@ def action
         # TODO deprecated, remove in v3.6
         return 'purchase!' if Workarea.config.auto_capture
 
-        if @order.items.all?(&:requires_shipping?)
-          Workarea.config.checkout_payment_action[:shipped]
-        elsif @order.items.any?(&:requires_shipping?)
-          Workarea.config.checkout_payment_action[:mixed]
+        if @order.items.all?(&:shipping?)
```

Comment:
this is still written in a way where it seems like there can only be one policy that ships.

Reply:
Yes, intentionally. This an acceptable compromise because the shipping policy has no logic, is decoratable, and there isn't any functionality that makes use of that distinction. For example bopus which is closest I can think of, doesn't need to define its own fulfillment policy.